client/components/filters/compare/index.js

```diff
@@ -49,6 +52,11 @@ class CompareFilter extends Component {
 		}
 	}
 
+	clearQuery() {
+		const { param, path, query } = this.props;
+		return getNewPath( { [ param ]: '' }, path, query );
```

Comment:
If the updated param is set to `undefined` instead of empty string, it will remove the parameter from the url so we don't get this

```
&product=
```